src/java_bytecode/java_types.h

```diff
@@ -611,10 +611,12 @@ class java_generic_symbol_typet : public symbol_typet
   /// \return The index of the type in the vector of generic types.
   optionalt<size_t> generic_type_index(const reference_typet &type) const
   {
-    const auto &type_pointer =
-      std::find(generic_types().begin(), generic_types().end(), type);
-    if(type_pointer != generic_types().end())
-      return type_pointer - generic_types().begin();
+    const auto type_variables = type.find(ID_type_variables);
+    for(std::size_t i = 0; i < generic_types().size(); ++i)
+    {
+      if(type_variables == generic_types()[i].find(ID_type_variables))
+        return i;
+    }
```

Comment:
This approach will do a lot of calls to `generic_types()`. At bare minimum create a temporary, but you might also do:
````c++
std::size_t index = 0;
for(const auto &generic_type : generic_types())
{
  if(type_variables == generic_type.find(ID_type_variables))
    return index;

  ++index;
}
````